### DISCUSSION TITLE
Migration for a specific context under specific circumstances

### DISCUSSION THREAD

**Discussion by avan2s** [prisma/prisma#13168] [Q&A] [answered]
Hello,

currently i'am evaluating prisma migration tool and i like it for typescript. I came from the java liquibase side and there were options to execute migrations only in a specific context or only if a specific pre condition is matching.

1. Specific Context: When starting the migration you can give a parameter in liquibase with many contexts "--contexts dev,test". If you deploy the migrations in contexts "prod", the migration tool did not execute this migration, because it didn't match any of the contexts. But in dev and test environment, it was executed. For maintaining different environments, this was really helpful. Currently the prisma migrate tool does not has such a parameter.

2. Pre-conditions: Another point are pre conditions. Sometimes it is neccessary for us to execute changes ONLY if a specific condition is matching,  If "SELECT COUNT(ID) FROM tenant where tenant_name="some tenant" = 1 ( => is available). then execute some specific sql statements. We used this i.e. for sql insert statements, to provide configuration records only for databases, where a specific tenant was present. Also we used this feature, to execute database scripts, that fixed some bugs in the database (like duplicate entries, encoding issues) caused by a bug in the application. So we didn't want to apply it to all databases, only under specific circumstances at the right time inside the migration process.

Is there some similar feature or other approach in prisma to realise such scenarios?

Thanks a lot for your help and best regards :)
Andreas

**Comment by janpio** [comment, marked as answer]
Both are not directly possible right now. 

Migrate only has one migration history that is executed on all environments your run it on (via switching of the `DATABASE_URL`).
There is also no flow control at all in these migrations (except what is possible in 1 SQL statement - which is minimal). Your description actually makes them sounds more like a tool for data migrations than schema migrations.

But both things seem interesting and valuable. I think you could create separate feature requests for them.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
